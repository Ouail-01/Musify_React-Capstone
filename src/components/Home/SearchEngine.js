import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';

export default function SearchEngine() {
  const [search, setSearch] = useState('');
  const navigate = useNavigate();

  const handleChange = (e) => {
    setSearch(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    navigate(`/searchResults/${search}`);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        value={search}
        onChange={handleChange}
        placeholder="Search..."
      />
    </form>
  );
}
