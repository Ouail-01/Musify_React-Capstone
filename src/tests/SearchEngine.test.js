import React from 'react';
import { render } from '@testing-library/react';
import SearchEngine from '../components/Home/SearchEngine';

describe('SearchEngine component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<SearchEngine />);
    expect(asFragment()).toMatchSnapshot();
  });
});
