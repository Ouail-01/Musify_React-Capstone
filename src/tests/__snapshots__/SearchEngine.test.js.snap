// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchEngine component renders correctly 1`] = `
<DocumentFragment>
  <form
    action="/searchResults/"
  >
    <input
      type="text"
    />
  </form>
</DocumentFragment>
`;
